### PR TITLE
[BugFix] Fix missing assertions in grouped GEMM tests

### LINKED ISSUE #198: [BugFix] Fix missing assertions in grouped GEMM tests
## Problem

Several test functions do not actually assert correctness:

1. **`tests/test_grouped_gemm_triton.py`** (lines 840-918): `check_kernel()` returns a boolean but callers never assert on it. A wrong output produces only a print, not a test failure.

2. **`tests/ops/test_grouped_gemm.py`** `test_grouped_gemm_complete` (lines 86-117): Only verifies gradients are not `None`, never compares output against a reference implementation.

## Impact

These tests can pass even when the kernels produce incorrect results.

## Fix

1. Change `check_kernel()` callers to `assert check_kernel(...)` or refactor to use `pytest` assertions internally.
2. Add correctness assertions in `test_grouped_gemm_complete` comparing output to `torch.bmm` or `ref_program`.

### PR DESCRIPTION
**Pull request by zhen8838**
Closes #198

## Summary
- Add `assert` before `check_kernel()` calls in `tests/test_grouped_gemm_triton.py` — previously the return value was silently ignored, so wrong kernel outputs never failed the test
- Remove `return success` from pytest test functions to eliminate `PytestReturnNotNoneWarning`
- Add forward+backward correctness check to `test_grouped_gemm_complete` in `tests/ops/test_grouped_gemm.py` using `test.check_fn()` with `requires_grad_(True)` on A and B inputs

## Test plan
- [x] `pre-commit run --all-files` passes
- [ ] CI `tileops_test_release` passes (pending)
- Note: `test_deepseek_nsa_gqa_window_sliding` failure is pre-existing (max err: 5.56), unrelated to this PR

### CODE REVIEW

File: tests/ops/test_grouped_gemm.py [hunk 1]
```diff
@@ -91,6 +91,7 @@ def test_grouped_gemm_complete(batch_sum: int, batch_count: int, N: int, K: int,
     benchmark = GroupedGemmBenchmark(batch_sum, batch_count, N, K, dtype)
 
     inputs = benchmark.gen_inputs()
```
[the remark this answers is not in the document]
Comment (zhen8838) [reply tests/ops/test_grouped_gemm.py]: Accepted. We now set `requires_grad_(True)` on A and B only (not on int32 batch tensors which can't have gradients), then call `check_fn` with the default `grad=True`. This properly tests both forward and backward correctness. See eb3e767.

File: tileops/kernels/mhc/mhc_pre.py [hunk 2]
```diff
@@ -276,6 +276,8 @@ def _(
     n_expand: int,
     c_x: int,
     dtype: str,
+    block_x_b: int,
+    block_C: int,
     num_stages: int,
     threads: int,
```
[the remark this answers is not in the document]
Comment (zhen8838) [reply tileops/kernels/mhc/mhc_pre.py:282]: Accepted. The mHC changes have been removed from this PR — they were incorrectly included. The mHC fix is tracked separately in PR #205. This PR now only touches `tests/ops/test_grouped_gemm.py` and `tests/test_grouped_gemm_triton.py`.

File: tileops/kernels/mhc/mhc_post.py [hunk 3]
```diff
@@ -79,6 +79,8 @@ def _(
     n_expand: int,
     c_x: int,
     dtype: str,
+    block_x_b: int,
+    block_C: int,
     num_stages: int,
     threads: int,
     *input,
```
[the remark this answers is not in the document]
Comment (zhen8838) [reply tileops/kernels/mhc/mhc_post.py:86]: Accepted. Same as above — the mHC changes have been removed from this PR. They belong to PR #205.
